modules/aidemBidAdapter.js

```diff
@@ -0,0 +1,691 @@
+import {
+  _each,
+  contains,
+  deepAccess,
+  deepSetValue,
+  getDNT,
+  isInteger,
+  logError,
+  logInfo,
+  isBoolean,
+  isStr
+} from '../src/utils.js';
+import {config} from '../src/config.js';
+import {BANNER, VIDEO} from '../src/mediaTypes.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {getRefererInfo} from '../src/refererDetection.js';
+import {ajax} from '../src/ajax.js';
+
+const BIDDER_CODE = 'aidem';
+const BASE_URL = 'https://zero.aidemsrv.com';
+const LOCAL_BASE_URL = 'http://127.0.0.1:8787';
+
+const AVAILABLE_CURRENCIES = ['USD'];
+const DEFAULT_CURRENCY = ['USD']; // NOTE - USD is the only supported currency right now; Hardcoded for bids
+const SUPPORTED_MEDIA_TYPES = [BANNER, VIDEO];
+const REQUIRED_VIDEO_PARAMS = [ 'mimes', 'protocols', 'context' ];
+
+export const ERROR_CODES = {
+  BID_SIZE_INVALID_FORMAT: 1,
+  BID_SIZE_NOT_INCLUDED: 2,
+  PROPERTY_NOT_INCLUDED: 3,
+  SITE_ID_INVALID_VALUE: 4,
+  MEDIA_TYPE_NOT_SUPPORTED: 5,
+};
+
+const endpoints = {
+  request: `${BASE_URL}/bid/request`,
+  notice: {
+    win: `${BASE_URL}/notice/win`,
+    timeout: `${BASE_URL}/notice/timeout`,
+    error: `${BASE_URL}/notice/error`,
+  }
+}
+
+export function setEndPoints(env = null, path = '', mediaType = BANNER) {
+  switch (env) {
+    case 'local':
+      endpoints.request = mediaType === BANNER ? `${LOCAL_BASE_URL}${path}/bid/request` : `${LOCAL_BASE_URL}${path}/bid/videorequest`
+      endpoints.notice.win = `${LOCAL_BASE_URL}${path}/notice/win`
+      endpoints.notice.error = `${LOCAL_BASE_URL}${path}/notice/error`
+      endpoints.notice.timeout = `${LOCAL_BASE_URL}${path}/notice/timeout`
+      break;
+    case 'main':
+      endpoints.request = mediaType === BANNER ? `${BASE_URL}${path}/bid/request` : `${BASE_URL}${path}/bid/videorequest`
+      endpoints.notice.win = `${BASE_URL}${path}/notice/win`
+      endpoints.notice.error = `${BASE_URL}${path}/notice/error`
+      endpoints.notice.timeout = `${BASE_URL}${path}/notice/timeout`
+      break;
+  }
+  return endpoints
+}
+
+let ortb2 = {}
+
+config.getConfig('aidem', function (config) {
+  if (config.aidem.env) { setEndPoints(config.aidem.env, config.aidem.path, config.aidem.mediaType) }
+})
+
+config.getConfig('ortb2', function (config) {
+  ortb2 = Object.assign({}, ortb2, config.ortb2)
+})
+
+// AIDEM Custom FN
+function recur(obj) {
+  var result = {}; var _tmp;
+  for (var i in obj) {
+    // enabledPlugin is too nested, also skip functions
+    if (!(i === 'enabledPlugin' || typeof obj[i] === 'function')) {
+      if (typeof obj[i] === 'object') {
```

Comment:
keep in mind that `typeof null === 'object'`.

Reply:
@dgirardi done.